test/step-function.test.ts

```diff
@@ -3912,20 +4000,24 @@ test("try { list.forEach(item => task(item)) }", () => {
 
 test("try { list.forEach(item => throw) }", () => {
   const { stack } = initStepFunctionApp();
-  const definition = new ExpressStepFunction(stack, "fn", (list: string[]) => {
-    try {
-      return list.forEach(() => {
-        throw new Error("cause");
-      });
-    } catch {
-      return null;
+  const definition = new ExpressStepFunction<{ list: string[] }, void | null>(
```

Comment:
do you prefer this way of modeling the function signature? I would much prefer it on the actual function. I think most people would prefer that also since it is more familiar?

```ts
(input: {list: string[]}): void | null => ..
```

Reply:
I actually don't think its more familiar because we generally do not need to put types on arrow function signatures unless necessary

I think it makes more sense to add it on the class because other types and methods may rely on the generic types than just the closure method.

That said, I am fine with either. For some reason I didn't think the inference would work, but it does.

```ts
  const definition = new ExpressStepFunction(
    stack,
    "fn",
    (input: { input: { id: string } }) => {
      return input.input.id;
    }
  ).definition;
```

Reply:
I think a developer is more likely to add it to the type parameters, but I am glad we support both.